stdlib/public/core/Diffing.swift

```diff
@@ -53,6 +53,9 @@ extension CollectionDifference {
   }
 }
 
+// Error type allows the use of throw to unroll state on application failure
+fileprivate enum _ApplicationError : Error { case failed }
```

Comment:
Nit: `fileprivate` at the file level is equivalent to `private`, which is preferred.